crates/ruff_python_semantic/src/scope.rs

```diff
@@ -38,9 +38,6 @@ pub struct Scope<'a> {
     /// In this case, the binding created by `x = 2` shadows the binding created by `x = 1`.
     shadowed_bindings: HashMap<BindingId, BindingId, BuildNoHashHasher<BindingId>>,
 
-    /// A list of all names that have been deleted in this scope.
-    deleted_symbols: Vec<&'a str>,
```

Comment:
A nice side-effect: this is no longer necessary. Now that we add to the scope when we hit a deletion, we don't have to track deleted symbols separately -- since every deleted symbol will appear in `bindings`.